skbio/diversity/beta/_unifrac.py

```diff
@@ -8,23 +8,17 @@
 
 from __future__ import absolute_import, division, print_function
 
+import numpy as np
+
 from skbio.util._decorator import experimental
 from skbio.diversity._base import (_validate_counts_vectors,
-                                   _validate_otu_ids_and_tree)
-
-
-def _observed_otu_counts(counts, otu_ids):
-    return {o: c for o, c in zip(otu_ids, counts) if c >= 1}
-
-
-def _validate(u_counts, v_counts, otu_ids, tree):
-    _validate_counts_vectors(u_counts, v_counts, suppress_cast=True)
-    _validate_otu_ids_and_tree(counts=u_counts, otu_ids=otu_ids, tree=tree)
+                                   _validate_otu_ids_and_tree,
+                                   _vectorize_counts_and_tree)
+from skbio.diversity._base_cy import _tip_distances
 
 
 @experimental(as_of="0.4.0-dev")
-def unweighted_unifrac(u_counts, v_counts, otu_ids, tree,
-                       validate=True, **kwargs):
+def unweighted_unifrac(u_counts, v_counts, otu_ids, tree, validate=True):
```

Comment:
Doc ``validate`` in the same way as for alpha. Also, be sure to be consistent about referring to samples versus count vectors.